### PR TITLE
Customize klusterlet namespace from klusterletConfig

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/bootstrap/render.go [hunk 1]
```diff
@@ -329,6 +331,32 @@ func filesToObjects(files []string, config interface{}) ([]runtime.Object, error
...
+func getKlusterletName(config *klusterletconfigv1alpha1.KlusterletConfig, clusterName, namespace string, mode operatorv1.InstallMode) string {
+	if mode == operatorv1.InstallModeHosted || mode == operatorv1.InstallModeSingletonHosted {
+		return fmt.Sprintf("klusterlet-%s", clusterName)
+	}
+	klusterletNamespace := getKlusterletNamespace(namespace, config)
+	if klusterletNamespace == constants.DefaultKlusterletNamespace {
+		return "klusterlet"
+	}
+	postFix := strings.TrimPrefix(klusterletNamespace, "open-cluster-management-")
+	return fmt.Sprintf("klusterlet-%s", postFix)
```
Comment (zhujian7) [inline pkg/bootstrap/render.go:357]: so the klusterlet name is determined by klusterlet namespace?
if the klusterlet namespace changes, we will delete the old klusterlet and create a new one?

Comment (zhujian7) [reply pkg/bootstrap/render.go:357]: the klustelet manifestwork delete option is orphan, so if the klusterlet namespace changes, there will be 2 klusterlet CRs on the managed cluster, right? I do not think that is what we expected.

**Comment by qiujian16** [comment]
before we merge this, do we also need to update mce installer with updated CRD at first @elgnay

**Comment by elgnay** [comment]
> before we merge this, do we also need to update mce installer with updated CRD at first @elgnay

Yes, we cannot merge this PR until the KlusterletConfig API is updated in the backplane-operator repo.

**Pull request merged**